src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodCapiDriver.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodCapiDriver : GpioDriver
+    {
+        private const string BasePath = "/dev/";
+
+        public string GpioChip { get; set; } = "gpiochip0";
+
+        private SafeChipHandle _chip;
+
+        private Dictionary<int, SafeLineHandle> _pinNumberToSafeLineHandle;
+
+        private timespec _defaultTimeOut = new timespec(IntPtr.Zero, 1_000_000); // one millisecond
```

Comment:
Looks like this line is not compiling. Where does timespec come from?

Reply:
@buyaa-n there is something wrong with the name of that file (is it included in the project?)

Reply:
oh i see, file name is not matching with class name, thanks @krwq